Prepared storage_pubsub_notifications snippet for inclusion on C.G.C

cbonnie opened the pull request:
<!-- Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to -->




<!--
Replace each [ ] with [X] to check it. Switch to the preview view to make it easier to click on links.
These steps will speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.
-->
If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement
    - release-note:bug
    - release-note:note
    - release-note:new-resource
    - release-note:new-datasource
    - release-note:deprecation
    - release-note:breaking-change
    - release-note:none
    
Unless you choose release-note:none, please add a release note.

See .ci/RELEASE_NOTES_GUIDE.md for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
```

modular-magician commented:
Oops! It looks like you're using an unknown release-note type in your changelog entries:
* REPLACEME

Please only use the types listed in https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md.

modular-magician commented:
Hello!  I am a robot who works on Magic Modules PRs.

I've detected that you're a community contributor. @rileykarson, a repository maintainer, has been assigned to assist you and help review your changes. 

<details>
  <summary>:question: First time contributing? Click here for more details</summary>

---

Your assigned reviewer will help review your code by: 
* Ensuring it's backwards compatible, covers common error cases, etc.
* Summarizing the change into a user-facing changelog note.
* Passes tests, either our "VCR" suite, a set of presubmit tests, or with manual test runs.

You can help make sure that review is quick by running local tests and ensuring they're passing in between each push you make to your PR's branch. Also, try to leave a comment with each push you make, as pushes generally don't generate emails.

If your reviewer doesn't get back to you within a week after your most recent change, please feel free to leave a comment on the issue asking them to take a look! In the absence of a dedicated review dashboard most maintainers manage their pending reviews through email, and those will sometimes get lost in their inbox.

---

</details>

modular-magician commented:
Oops! It looks like you're using an unknown release-note type in your changelog entries:
* REPLACEME

Please only use the types listed in https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md.

modular-magician commented:
Oops! It looks like you're using an unknown release-note type in your changelog entries:
* REPLACEME

Please only use the types listed in https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md.

modular-magician commented:
Oops! It looks like you're using an unknown release-note type in your changelog entries:
* REPLACEME

Please only use the types listed in https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md.

cbonnie commented:
Sorry for writing so many commits without descriptions, I'm new to GitHub and I keep forgetting to do that. The commits are all small typo-like fixes (removed a rogue "#", added periods to comments, changed a comment to describe the code).

rileykarson commented on mmv1/products/cgc/terraform.yaml:
```diff
@@ -555,6 +555,18 @@ overrides: !ruby/object:Overrides::ResourceOverrides
+      - !ruby/object:Provider::Terraform::Examples
+        name: "main"
+        primary_resource_id: "topic"
+        primary_resource_type: "google_pubsub_topic"
```
```suggestion
```

This one can generally be omitted

rileykarson commented on mmv1/products/cgc/terraform.yaml:
```diff
@@ -555,6 +555,18 @@ overrides: !ruby/object:Overrides::ResourceOverrides
+          example_bucket_name: "example-bucket-name"
+          your_topic_name: "your_topic_name"
+        min_version: beta
+        # ignore_read_extra:
```
Can you delete these commented out lines?

cbonnie replied:
Done

rileykarson commented on mmv1/templates/terraform/examples/storage_pubsub_notifications.tf.erb:
```diff
@@ -0,0 +1,32 @@
+}
+
+// Create a Pub/Sub topic.
+resource "google_pubsub_topic" "<%= ctx[:primary_resource_id] %>" {
```
Is this meant to be an IAM resource?

rileykarson commented:
Handing off review to @c2thorn since I'll be out for a while in a day or so!

cbonnie commented on mmv1/products/cgc/terraform.yaml:
```diff
@@ -551,10 +551,13 @@ overrides: !ruby/object:Overrides::ResourceOverrides
-        #   - "target"
-        #   - "ip_address"
+      - !ruby/object:Provider::Terraform::Examples
+        name: "main"
```
Done

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6031-old..auto-pr-6031) ( 1 file changed, 58 insertions(+))
TF Cloud Doc Samples: [Diff](https://github.com/modular-magician/terraform-docs-samples/compare/auto-pr-6031-old..auto-pr-6031) ( 1 file changed, 32 insertions(+))

modular-magician commented:
#### Tests analytics 
Total tests: `2038` 
Passed tests `1804` 
Skipped tests: `231` 
Failed tests: `3` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary> TestAccContainerCluster_withConfidentialNodes|TestAccContainerCluster_withAddons|TestAccCGCSnippet_storagePubsubNotificationsExample </details>

modular-magician commented:
Tests failed during RECORDING mode:
 `TestAccCGCSnippet_storagePubsubNotificationsExample`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6031/artifacts/abe4cba4-6f04-4d93-9c82-1ea1a0d8b8ce/recording/TestAccCGCSnippet_storagePubsubNotificationsExample.log)]
`TestAccContainerCluster_withConfidentialNodes`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6031/artifacts/abe4cba4-6f04-4d93-9c82-1ea1a0d8b8ce/recording/TestAccContainerCluster_withConfidentialNodes.log)]
`TestAccContainerCluster_withAddons`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6031/artifacts/abe4cba4-6f04-4d93-9c82-1ea1a0d8b8ce/recording/TestAccContainerCluster_withAddons.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6031/artifacts/abe4cba4-6f04-4d93-9c82-1ea1a0d8b8ce/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6031/artifacts/abe4cba4-6f04-4d93-9c82-1ea1a0d8b8ce/recording) for each test

c2thorn commented on mmv1/templates/terraform/examples/storage_pubsub_notifications.tf.erb:
```diff
@@ -0,0 +1,32 @@
+
+// Create a Pub/Sub topic.
+resource "google_pubsub_topic_iam_binding" "binding" {
+  name    = "default_iam_binding"
```
The test is failing because the iam binding doesn't have a `name` field
```suggestion
```

modular-magician commented:
#### Tests analytics 
Total tests: `2038` 
Passed tests `1804` 
Skipped tests: `231` 
Failed tests: `3` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary> TestAccContainerCluster_withConfidentialNodes|TestAccContainerCluster_withAddons|TestAccCGCSnippet_storagePubsubNotificationsExample </details>

modular-magician commented:
Tests failed during RECORDING mode:
 `TestAccCGCSnippet_storagePubsubNotificationsExample`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6031/artifacts/2eb0e64b-03e5-4aa7-94cd-941cbe1380b6/recording/TestAccCGCSnippet_storagePubsubNotificationsExample.log)]
`TestAccContainerCluster_withConfidentialNodes`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6031/artifacts/2eb0e64b-03e5-4aa7-94cd-941cbe1380b6/recording/TestAccContainerCluster_withConfidentialNodes.log)]
`TestAccContainerCluster_withAddons`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6031/artifacts/2eb0e64b-03e5-4aa7-94cd-941cbe1380b6/recording/TestAccContainerCluster_withAddons.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6031/artifacts/2eb0e64b-03e5-4aa7-94cd-941cbe1380b6/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6031/artifacts/2eb0e64b-03e5-4aa7-94cd-941cbe1380b6/recording) for each test

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6031-old..auto-pr-6031) ( 1 file changed, 57 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6031-old..auto-pr-6031) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF Cloud Doc Samples: [Diff](https://github.com/modular-magician/terraform-docs-samples/compare/auto-pr-6031-old..auto-pr-6031) ( 1 file changed, 31 insertions(+))

modular-magician commented:
#### Tests analytics 
Total tests: `2039` 
Passed tests `1805` 
Skipped tests: `231` 
Failed tests: `3` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary> TestAccContainerCluster_withConfidentialNodes|TestAccContainerCluster_withAddons|TestAccCGCSnippet_storagePubsubNotificationsExample </details>

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccCGCSnippet_storagePubsubNotificationsExample`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6031/artifacts/49857d6a-2763-4f70-893b-465e65e652c9/recording/TestAccCGCSnippet_storagePubsubNotificationsExample.log)] 

Tests failed during RECORDING mode:
 `TestAccContainerCluster_withConfidentialNodes`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6031/artifacts/49857d6a-2763-4f70-893b-465e65e652c9/recording/TestAccContainerCluster_withConfidentialNodes.log)]
`TestAccContainerCluster_withAddons`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6031/artifacts/49857d6a-2763-4f70-893b-465e65e652c9/recording/TestAccContainerCluster_withAddons.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6031/artifacts/49857d6a-2763-4f70-893b-465e65e652c9/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6031/artifacts/49857d6a-2763-4f70-893b-465e65e652c9/recording) for each test
